packages/broker/test/unit/plugins/operator/InspectRandomNodeService.test.ts

```diff
@@ -0,0 +1,281 @@
+        helper.getOperatorsInSponsorship.mockImplementation(async (sponsorshipAddress) => {
+            return sponsorships.find(({ address }) => address === sponsorshipAddress)!.operators
+        })
+    }
+
+    function setStreamPartsAssignedToMe(streamParts: StreamPartID[]) {
+        loadBalancer.getPartitionsOfStreamAssignedToMe.mockImplementation((streamId) => {
+            return streamParts.filter((streamPart) => StreamPartIDUtils.getStreamID(streamPart) === streamId)
+        })
+        loadBalancer.isAnyPartitionOfStreamAssignedToMe.mockImplementation((streamId) => {
+            return streamParts.some((streamPart) => StreamPartIDUtils.getStreamID(streamPart) === streamId)
+        })
+    }
+
+    beforeEach(() => {
+        helper = mock<InspectRandomNodeHelper>()
+        loadBalancer = mock<StreamAssignmentLoadBalancer>()
+    })
+
+    it('returns undefined if no sponsorships are found', async () => {
+        setupEnv([])
+        setStreamPartsAssignedToMe([])
+        const result = await findTarget(MY_OPERATOR_ADDRESS, helper, loadBalancer)
+        expect(result).toBeUndefined()
+    })
+
+    it('returns undefined if no sponsorships found with an operatorCount >= 2', async () => {
+        setupEnv([{
+            address: SPONSORSHIP_ADDRESS,
+            operators: [MY_OPERATOR_ADDRESS],
+            streamId: STREAM_ID,
+        }])
+        setStreamPartsAssignedToMe([])
+        const result = await findTarget(MY_OPERATOR_ADDRESS, helper, loadBalancer)
+        expect(result).toBeUndefined()
+    })
+
+    it('returns undefined if no sponsorships found with a partition assigned to me', async () => {
+        setupEnv([{
+            address: SPONSORSHIP_ADDRESS,
+            operators: [MY_OPERATOR_ADDRESS, OTHER_OPERATOR_ADDRESS],
+            streamId: STREAM_ID,
+        }])
+        setStreamPartsAssignedToMe([])
+        const result = await findTarget(MY_OPERATOR_ADDRESS, helper, loadBalancer)
+        expect(result).toBeUndefined()
+    })
+
+    it('returns target sponsorship, operator and stream part', async () => {
+        setupEnv([{
+            address: SPONSORSHIP_ADDRESS,
+            operators: [MY_OPERATOR_ADDRESS, OTHER_OPERATOR_ADDRESS],
+            streamId: STREAM_ID,
+        }])
+        setStreamPartsAssignedToMe([
+            toStreamPartID(STREAM_ID, 0),
+            toStreamPartID(STREAM_ID, 1),
+            toStreamPartID(STREAM_ID, 2),
+        ])
+
+        const result = await findTarget(MY_OPERATOR_ADDRESS, helper, loadBalancer)
+        expect(result).toMatchObject({
+            sponsorshipAddress: SPONSORSHIP_ADDRESS,
+            operatorAddress: OTHER_OPERATOR_ADDRESS,
+            streamPart: expect.stringMatching(/^streamId#\d$/)
+        })
+    })
+
+    // TODO: few edge-cases where state changes during asynchronicity
+})
+
+const PEER_DESCRIPTOR_ONE = { id: 'node-1' }
+const PEER_DESCRIPTOR_TWO = { id: 'node-2' }
+const PEER_DESCRIPTOR_THREE = { id: 'node-3' }
```

Comment:
Not relevant in this test, but maybe peer descriptors could be hex values: easier to see that these values are node ids

```suggestion
const PEER_DESCRIPTOR_ONE = { id: '0x1111' }
const PEER_DESCRIPTOR_TWO = { id: '0x2222' }
const PEER_DESCRIPTOR_THREE = { id: '0x3333' }
```

Reply:
fixed